src/nvim/screen.c

```diff
@@ -5461,7 +5461,7 @@ static void win_redr_border(win_T *wp)
   int *attrs = wp->w_float_config.border_attr;
 
   int *adj = wp->w_border_adj;
-  int irow = wp->w_height_inner, icol = wp->w_width_inner;
+  int irow = wp->w_height, icol = wp->w_width;
```

Comment:
`w_height` and `w_width` are not "source of truth" variables here. Should be `wp->w_height_inner + wp->w_winbar_height` and `wp->w_width_inner` for now. (we will need to rethink this more when adding support for even more embedded widgets, like scrollbar).

Reply:
> Should be `wp->w_height_inner + wp->w_winbar_height` and `wp->w_width_inner` for now.

Actually this is also what is first thought.